### PR TITLE
proof-of-concept support for coroutines

### PR DESCRIPTION
**Pull request by evaleev**

### CODE REVIEW

File: ttg/ttg/madness/ttg.h [hunk 1]
```diff
@@ -298,28 +304,83 @@ namespace ttg_madness {
...
+
+          // resume the coroutine
+          auto ret = static_cast<ttg::resumable_task>(ttg::coroutine_handle<>::from_address(suspended_task_address));
+          assert(ret.ready());
+          ret.resume();
+          if (ret.completed()) {
+            ret.destroy();
+            suspended_task_address = nullptr;
+          } else {  // not yet completed
+            abort();
```
Comment (devreal) [inline ttg/ttg/madness/ttg.h]: Why abort here? We should use `ttg::abort` instead to avoid nasty hangs if we ever hit this

Comment (evaleev) [reply ttg/ttg/madness/ttg.h]: @devreal this is a safety-check abort(), but I took this opportunity to replace more std::abort() instances with ttg::abort()

**Review by devreal — APPROVED** [review]
I will need to spend more time understanding the details but let's merge into my branch first. Thanks @evaleev!

**Comment by evaleev** [comment]
@devreal OK, go ahead and merge then, I can't seem to be able to

**Comment by devreal** [comment]
@evaleev CI fails saying: "Cannot Compile simple program using std::coroutine". GCC 10 should be sufficient no?

**Comment by evaleev** [comment]
> @evaleev CI fails saying: "Cannot Compile simple program using std::coroutine". GCC 10 should be sufficient no?

Discovery of coroutine support is unfortunately trickier than I assumed ... presence of header is not enough, etc. I tried to revise, fingers crossed.

Release builds should succeed, but not the debug builds due to the termdet issue I communicated via slack.

**Comment by evaleev** [comment]
OK, aside from the (orthogonal) termdet issue CI passes now

**Comment by devreal** [comment]
Thanks @evaleev ! Looks like I have to approve every update to run on CI. We'll investigate the termdet issue once the SC dust settles.

**Comment by evaleev** [comment]
you can see original (source) branch's CI results in the main repo: https://github.com/TESSEorg/ttg/actions/runs/3425628942

p.s. for small teams single repo (no forks) make most sense by reducing friction and noise

**Pull request merged**